### DISCUSSION TITLE
Get active menu title

### DISCUSSION THREAD

**Discussion by Fedora-Man** [holger1411/understrap#1741] [Q&A]
How to get active menu title/path to display in the current page.

**Comment by bacoords** [comment]
If you're using the WordPress nav_menu functions, you should be able to see the classes on the list items: `current-menu-item` and `current_page_item` and `active` are some examples.

**Reply by Fedora-Man** [reply]
I am not trying to manipulate the css classes but I want to display the active menu item text in a header banner.

**Reply by bacoords** [reply]
Ah I see - I suppose you could write your own Navwalker that basically ignores everything that's not the active menu item? This could be tricky. 

Or you could use a function like `wp_get_nav_menu_object()` to get the actual menu object and loop through it that way.

**Comment by IanDelMar** [comment]
Assuming the menu location is "primary" put this into your functions.php:

```php
function slug_get_nav_menu( $name ) {
	$locations = get_nav_menu_locations();
	if ( ! isset( $locations[ $name ] ) ) {
		return false;
	}

	return wp_get_nav_menu_object( $locations[ $name ] );
}

function slug_get_current_menu_item_title() {
	$menu = slug_get_nav_menu( 'primary' );
	if ( false === $menu ) {
		return '';
	}

	$nav_menu_items = wp_get_nav_menu_items( $menu );
	$current_item   = current( wp_filter_object_list( $nav_menu_items, array( 'object_id' => get_queried_object_id() ) ) );

	return false !== $current_item ? $current_item->title : '';
}
```

Then echo `slug_get_current_menu_item_title()` wherever you want the menu title to appear. Note that this won't work on pages that do not have an ID such as archives.
